algorithms/src/polycommit/kzg10/mod.rs

```diff
@@ -401,16 +398,38 @@ impl<E: PairingEngine> KZG10<E> {
             None
         };
 
-        Ok(Proof { w: w.to_affine(), random_v })
+        Ok(KZGProof { w: w.to_affine(), random_v })
+    }
+
+    /// On input a polynomial `p` in Lagrange basis, and a point `point`,
+    /// outputs an evaluation proof for the same.
+    pub fn open_lagrange(
+        lagrange_basis: &LagrangeBasis<E>,
+        domain_elements: &[E::Fr],
+        evaluations: &[E::Fr],
+        point: E::Fr,
+        evaluation_at_point: E::Fr,
+    ) -> Result<KZGProof<E>, PCError> {
+        Self::check_degree_is_too_large(evaluations.len() - 1, lagrange_basis.size())?;
+        assert_eq!(evaluations.len().next_power_of_two(), lagrange_basis.size());
+        let mut divisor_evals = cfg_iter!(domain_elements).map(|&e| e - point).collect::<Vec<_>>();
+        snarkvm_fields::batch_inversion(&mut divisor_evals);
+        cfg_iter_mut!(divisor_evals).zip(evaluations).for_each(|(divisor_eval, &eval)| {
```

Comment:
should the length of `divisor_evals` be equal to that of `evaluations`?

Reply:
Changed to `zip_eq` in [`a2f44b5` (#1100)](https://github.com/AleoHQ/snarkVM/pull/1100/commits/a2f44b5d4ea8d5120009f808ec77a853cab3daa8)